flood_api/utils/json_utilities.py

```diff
@@ -0,0 +1,57 @@
+import geopandas as gpd
+import pandas as pd
+from datetime import date
+import json
+
+def dataframe_to_geojson(
+    df: gpd.GeoDataFrame, 
+    columns: list[str], 
+    date_handler: callable = None, 
+    sort_columns: list[str] = None
+) -> dict:
+    """
+    Convert a GeoDataFrame to a GeoJSON.
+
+    Parameters:
+    - df (GeoDataFrame): The GeoDataFrame to convert.
+    - columns (list): The columns to include in the GeoJSON.
+    - date_handler (callable): The date handler to use for date columns.
+    - sort_columns (list): The columns to sort by.
+
+    Returns:
+    dict: The GeoJSON.
+    """
+    # Check if the dataframe is empty. If it is, return an 
+    # empty GeoJSON where we only assume the geometry column
+    # is present.
+    if df.empty:
+        return json.loads(df.to_json())
+    
+    # Sort the columns if a list of columns is provided
+    if sort_columns is not None:
+        df = df.sort_values(by=sort_columns)
+
+    # Reset the index to make response cleaner
+    df = df.reset_index(drop=True)
+
+    # Serialize the dataframe as a string
+    geojson_as_string = df[columns + ['geometry']].to_json(default=date_handler)
```

Comment:
Could this always be `custom_date_handler`? Or will we ever need to be able to specify another handler?